File: src/nl/rubensten/texifyidea/LatexFileType.java
package nl.rubensten.texifyidea;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * @author Sten Wessel
 */
public class LatexFileType extends LanguageFileType {

    public static final LatexFileType INSTANCE = new LatexFileType();

    private LatexFileType() {
        super(LatexLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "LaTeX source file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "LaTeX source file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "tex";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return TexifyIcons.LATEX_FILE;
    }
}

File: src/nl/hannahsten/texifyidea/file/LatexFileType.kt
package nl.hannahsten.texifyidea.file

import com.intellij.openapi.fileTypes.LanguageFileType
import nl.hannahsten.texifyidea.grammar.LatexLanguage
import nl.hannahsten.texifyidea.TexifyIcons

/**
 * @author Sten Wessel
 */
object  LatexFileType : LanguageFileType(LatexLanguage) {

    override fun getName() = "LaTeX source file"

    override fun getDescription() = "LaTeX source file"

    override fun getDefaultExtension() = "tex"

    override fun getIcon() = TexifyIcons.LATEX_FILE

    // Fix https://github.com/Hannah-Sten/TeXiFy-IDEA/issues/2195#issuecomment-1024344147
    override fun getDisplayName() = name
}
